lib/livebook/notebook/explore/intro_to_nx.livemd

```diff
@@ -1,3 +1,732 @@
+that support specialized processors to speed up numeric computation including
+TPUs and GPUs.
+
+To know Nx, we'll get to know tensors first. This rapid overview will touch
+on the major libraries. Then, future notebooks will take a deep dive into working
+with tensors in detail, autograd, and backends. Then, we'll dive into specific
+problem spaces like Axon, the machine learning library.
+
+## Nx and tensors
+
+Systems of equations are a central theme in numerical computing.
+These equations are often expressed and solved with multidimensional
+arrays. For example, this is a two dimensional array:
+
+<!-- livebook:{"break_markdown":true} -->
+
+$$
+\begin{bmatrix}
+  1 & 2 \\
+  3 & 4
+\end{bmatrix}
+$$
+
+<!-- livebook:{"break_markdown":true} -->
+
+Elixir programmers typically express a similar data structure using
+a list of lists, like this:
+
+```elixir
+[
+  [1, 2],
+  [3, 4]
+]
+```
+
+This data structure works fine within many functional programming
+algorithms, but breaks down with deep nesting and random access.
+
+On top of that, Elixir numeric types lack optimization for many numerical
+applications. They work fine when programs
+need hundreds or even thousands of calculations. They tend to break
+down with traditional STEM applications when a typical problem
+needs millions of calculations.
+
+In Nx, we express multi-dimensional data using typed tensors. Simply put,
+a tensor is a multi-dimensional array with a predetermined shape and
+type. To interact with them, Nx relies on tensor-aware operators rather
+than `Enum.map/2` and `Enum.reduce/3`.
+
+In this section, we'll look at some of the various tools for
+creating and interacting with tensors.
+
+<!-- livebook:{"break_markdown":true} -->
+
+To get started, we'll first need to include the Nx dependency.
+In Livebook, we can simply use Mix install. At the time of this writing,
+there's no Hex package for Nx, so we'll use a git
+dependency, like this:
+
+```elixir
+Mix.install([
+  {:exla, "~> 0.1.0-dev", github: "elixir-nx/nx", sparse: "exla"},
+  {:nx, "~> 0.1.0-dev", github: "elixir-nx/nx", sparse: "nx", override: true}
+])
+```
+
+The IEx helpers will assist our exploration of the core tensor concepts.
+
+```elixir
+import IEx.Helpers
+```
+
+Now, everything is set up, so we're ready to create some tensors.
+
+<!-- livebook:{"break_markdown":true} -->
+
+### Creating tensors
+
+<!-- livebook:{"break_markdown":true} -->
+
+Start out by getting a feel for Nx through its documentation.
+Do so through the IEx helpers, like this:
+
+```elixir
+h(Nx)
+```
+
+Immediately, you can see that tensors are at the center of the
+API. The main API for creating tensors is `Nx.tensor/2`:
+
+```elixir
+h(Nx.tensor())
+```
+
+We use it to create tensors from raw Elixir lists of numbers, like this:
+
+```elixir
+tensor =
+  1..4
+  |> Enum.chunk_every(2)
+  |> Nx.tensor(names: [:y, :x])
+```
+
+We have a new tensor. Create your own tensor using a binary
```

Comment:
Should we describe what is the inspected result first? Something like, "we got a Nx.Tensor back, the first row declares the type of the tensor (s64) and the shape, followed by an Elixir-like representation of its contents, while in practice, the tensor is a blob of memory". Then I would show `Nx.to_binary/1`. Once we show the result of `Nx.to_binary/1`, we can ask the user to try Nx.from_binary by themselves. :) Otherwise I think it is a big leap for them to try.

WDYT?

Reply:
just a quick note to say this is perfect feedback... exactly what i needed. I will get to it all in the next few days.

Reply:
Revised this with a better discussion of binaries and a better flow. wdyt?